motionplan/tpSpaceRRT.go

```diff
@@ -900,28 +905,19 @@ func (mp *tpSpaceRRTMotionPlanner) smoothPath(ctx context.Context, path []node)
 		// get start node of first edge. Cannot be either the last or second-to-last node.
 		// Intn will return an int in the half-open interval half-open interval [0,n)
 		firstEdge := mp.randseed.Intn(len(path) - 2)
-		secondEdge := firstEdge + 2 + mp.randseed.Intn((len(path)-2)-firstEdge)
+		secondEdge := firstEdge + 1 + mp.randseed.Intn((len(path)-1)-firstEdge)
 
 		newInputSteps, err := mp.attemptSmooth(ctx, path, firstEdge, secondEdge, smoothPlanner)
 		if err != nil || newInputSteps == nil {
 			continue
 		}
 		newCost := sumCosts(newInputSteps)
 		if newCost >= currCost {
+			// The smoothed path is longer than the original
 			continue
 		}
-		// Re-connect to the final goal
-		if newInputSteps[len(newInputSteps)-1] != path[len(path)-1] {
```

Comment:
I like the simplification but I'm not sure why we dont need to do this anymore

Reply:
As far as I can tell, this was accounting for a bug that no longer exists.